packages/dcos-launch/extra/launch/config.py

```diff
@@ -49,375 +21,256 @@ def expand_path(path: str, relative_dir: str) -> str:
-    # Furthermore, we cannot use the target ssh_key_path as it will automatically
-    # invoked the validate fn from ssh/validate.py Luckily, we can instead use
-    # the more idiomatic 'ssh_private_key_filename'
-    targets.append(Target({
-        'ssh_user',
-        'ssh_port',
-        'master_list',
-        'agent_list',
-        'public_agent_list',
-        'ssh_parallelism',
-        'process_timeout'}))
-
-    resolver = resolve_configuration(sources, targets)
-    status = resolver.status_dict
-    if status['status'] == 'errors':
-        raise AssertionError(pretty_print_validate_error(status['errors'], status['unset']))
-
-
-def calculate_dcos_config_contents(dcos_config, num_masters, ssh_user, ssh_private_key, platform):
-    """ Fills in the ssh user, ssh private key and ip-detect script if possible
-    Takes the config's local references and converts them into transmittable content
-    """
-    user_config = yaml.load(dcos_config)
-    # Use the default config in the installer for the same experience
-    # w.r.t the auto-filled settings
-    config = yaml.load(dcos_installer.config.config_sample)
-    config.update(user_config)
-    config['ssh_user'] = ssh_user
-    config['ssh_key'] = ssh_private_key
-    for key_name in ['ip_detect_filename', 'ip_detect_public_filename']:
-        if key_name in config:
-            config[key_name.replace('_filename', '_contents')] = load_string(config[key_name])
-            del config[key_name]
-    if 'ip_detect_contents' not in config:
-        config['ip_detect_contents'] = test_util.helpers.ip_detect_script(platform)
-    return yaml.dump(config)
-
-
-def validate_onprem_provider_platform(provider, platform):
-    # TODO DCOS-14033: [gen.internals] Source validate functions are global only
-    if provider == 'onprem':
-        assert platform != 'azure', '`provider: onprem` is not currently not support on azure'
-    else:
-        assert platform == provider
+def load_ssh_private_key(doc):
+    if doc.get('key_helper') == 'true':
+        return 'unset'
+    if 'ssh_private_key_filename' not in doc:
+        return launch.util.NO_TEST_FLAG
+    return launch.util.load_string(doc['ssh_private_key_filename'])
 
 
-def validate_key_helper_support(platform, key_helper):
-    # TODO DCOS-14033: [gen.internals] Source validate functions are global only
-    if key_helper == 'false':
-        return
-    assert platform in ('aws', 'azure')
+class LaunchValidator(cerberus.Validator):
+    """ Needs to use unintuitive pattern so that child validator can be created
+    for validated the nested dcos_config
+    """
+    def __init__(self, *args, **kwargs):
+        super(LaunchValidator, self).__init__(*args, **kwargs)
+        if 'config_dir' in kwargs:
+            self.config_dir = kwargs['config_dir']
 
+    def _normalize_coerce_expand_local_path(self, value):
+        return expand_path(value, self.config_dir)
```

Comment:
The only way that would happen is if someone made a programming error. Config path (or alternatively, the info path) must always exist for any dcos-launch command to work. Thus, a committer would have to add logic that could allow the config or dir to be None. I could add an assertion to the class instantiator to make that clear